infrastructure/src/main/resources/schema.sql

```diff
@@ -0,0 +1,105 @@
+(
+    id       bigint unsigned not null auto_increment,
+    vorname  varchar(50)     not null,
+    nachname varchar(50)     not null,
+    anrede   varchar(10)     not null,
+    primary key (id)
+);
+
+create table if not exists veranstaltung
+(
+    id       bigint unsigned not null auto_increment,
+    name     varchar(50)     not null,
+    semester integer,
+    primary key (id)
+);
+
+create table if not exists student
+(
+    id       bigint unsigned not null auto_increment,
+    username varchar(50)     not null,
+    vorname  varchar(50),
+    nachname varchar(50),
+    primary key (username)
+);
+
+create table if not exists fragebogen
+(
+    id            bigint unsigned not null auto_increment,
+    name          varchar(100)    not null,
+    status        integer         not null,
+    startzeit     datetime,
+    endzeit       datetime        not null,
+    veranstaltung bigint unsigned,
+    einheit       enum ('VORLESUNG','UEBUNG','AUFGABE','PRAKTIKUM','DOZENT','BERATUNG','GRUPPE'),
+    primary key (id),
+    foreign key (veranstaltung)
+        references veranstaltung (id)
+);
+
+create table if not exists frage
+(
+    id         bigint unsigned not null auto_increment,
+    fragebogen bigint unsigned not null,
+    fragetext  varchar(100)    not null,
+    primary key (id),
+    foreign key (fragebogen)
+        references fragebogen (id)
+);
+
+create table if not exists antwort
+(
+    id          bigint unsigned not null auto_increment,
+    frage       bigint unsigned not null,
+    textantwort text,
+    primary key (id),
+    foreign key (frage)
+        references frage (id)
+);
+
+create table if not exists auswahl
+(
+    id          bigint unsigned not null auto_increment,
+    auswahltext varchar(100)    not null,
+    frage       bigint unsigned,
+    antwort     bigint unsigned,
+    primary key (id),
+    foreign key (frage)
+        references frage (id),
+    foreign key (antwort)
+        references antwort (id)
+);
+
+create table if not exists studentBeantwortetFragebogen
+(
+    student    varchar(50)     not null,
+    fragebogen bigint unsigned not null,
+    primary key (student, fragebogen),
+    foreign key (student)
+        references student (id),
+    foreign key (fragebogen)
+        references fragebogen (id)
+);
+
+create table if not exists studentBelegtVeranstaltung
+(
+    student       varchar(50)     not null,
+    veranstaltung bigint unsigned not null,
+    primary key (student, veranstaltung),
+    foreign key (student)
+        references student (id),
+    foreign key (veranstaltung)
+        references veranstaltung (id)
+);
+
+create table if not exists dozentOrganisiertVeranstaltung
+(
+    dozent        bigint unsigned not null,
+    veranstaltung bigint unsigned not null,
+    primary key (dozent, veranstaltung),
+    foreign key (dozent)
+        references dozent (id),
+    foreign key (veranstaltung)
+        references veranstaltung (id)
+);
```

Comment:
We don't have an m:n relationship between lecturer and event. We have an 1:n relationship (one lecturer has n events, but one event has only one lecturer).

Reply:
It's possible that more than one 'Dozent' wants to edit a Lecture. In reality the 'dozent' might not be the one to create the lecture or add the questionaires. It should be possible for other organisators to have the same rights and this is why we should use a n:m relationship here.

Reply:
Like for example if a lecture has multiple 'einheiten' like vorlesung, praktikum, uebung. And each of those will be organized by another person. These organisators should be able to make changes to the lecture as well.

Reply:
But then we have to change this in the business logic too. We only save the lecturer in the event. So we need a List of lecturers instead.